### PR TITLE
Update DataMigration SDK to use newest swagger api version

### PR DESCRIPTION
**Pull request by huang91shu**
This PR is based on the rest PR:
https://github.com/Azure/azure-rest-api-specs/pull/3567

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataMigration/DataMigration.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -9,11 +9,11 @@
 // General Information about an assembly is controlled through the following 
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
-[assembly: AssemblyTitle("DataMigration.Tests")]
```
Comment (dsgouda) [inline src/SDKs/DataMigration/DataMigration.Tests/Properties/AssemblyInfo.cs]: Undo these changes

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs [hunk 2]
```diff
@@ -26,6 +26,7 @@ public void CreateResourceSucceeds()
                 var service = CreateDMSInstance(context, dmsClient, resourceGroup, DmsDeploymentName);
                 var project = CreateDMSProject(context, dmsClient, resourceGroup, service.Name, DmsProjectName);
             }
+
```
Comment (dsgouda) [inline src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: Undo these changes

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDTaskTests.cs [hunk 3]
```diff
@@ -46,6 +50,49 @@ public void GetResourceSucceeds()
...
+            {
+                var resourceGroup = CreateResourceGroup(context, resourcesHandler, ResourceGroupName, TestConfiguration.Location);
+                var dmsClient = Utilities.GetDataMigrationManagementClient(context, dmsClientHandler);
+                var service = CreateDMSInstance(context, dmsClient, resourceGroup, DmsDeploymentName);
+                var project = CreateDMSProject(context, dmsClient, resourceGroup, service.Name, DmsProjectName);
+                var task = CreateDMSSyncTask(context, dmsClient, resourceGroup, service, project.Name, DmsTaskName);
+                bool wait = true;
+                do
+                {
+                    var getResult = dmsClient.Tasks.Get(resourceGroup.Name, service.Name, project.Name, task.Name, "output");
```
Comment (dsgouda) [inline src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDTaskTests.cs]: nit: is there an async version of these calls

File: src/SDKs/DataMigration/Management.DataMigration/Generated/DataMigrationServiceClient.cs [hunk 4]
```diff
@@ -46,30 +46,26 @@ public partial class DataMigrationServiceClient : ServiceClient<DataMigrationSer
         /// </summary>
         public ServiceClientCredentials Credentials { get; private set; }
 
-        /// <summary>
-        /// Version of the API
-        /// </summary>
-        public string ApiVersion { get; private set; }
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Generated/DataMigrationServiceClient.cs]: Why is this being deleted?

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 5]
```diff
@@ -17,6 +17,9 @@
 	</PropertyGroup>
 	<PropertyGroup>
 		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+		<AssemblyVersion>0.4.0.0</AssemblyVersion>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Please remove these lines
`AssemblyVersion` and `AssemblyFileVersion` must be set in the `AssemblyFile.cs` file similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/Properties/AssemblyInfo.cs)

File: src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -4,14 +4,14 @@
 using System.Reflection;
 using System.Resources;
 
-[assembly: AssemblyTitle("Microsoft Azure Data Migration Management Library")]
-[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Data Migration Resources.")]
+//[assembly: AssemblyTitle("Microsoft Azure Data Migration Management Library")]
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs]: Please undo these changes

File: src/SDKs/DataMigration/Management.DataMigration/generate.cmd [hunk 7]
```diff
@@ -0,0 +1,7 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd datamigration/resource-manager %*
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/generate.cmd]: Code must be generated using the generate.ps1, please remove this file

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please fix the CI failures too

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj [hunk 8]
```diff
@@ -2,13 +2,13 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
   <PropertyGroup>
     <PackageId>DataMigration.Tests</PackageId>
-    <VersionPrefix>1.0.0-preview</VersionPrefix>
-    <AssemblyName>DataMigration.Tests</AssemblyName>
     <Description>DataMigration.Tests Class Library</Description>
+    <AssemblyName>DataMigration.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
[the remark this answers is not in the document]
Comment (huang91shu) [reply src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: Done.

File: src/SDKs/DataMigration/DataMigration.Tests/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/DataMigration.Tests/Properties/AssemblyInfo.cs]: Done.

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs (hunk 2 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj (hunk 5 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/generate.cmd (hunk 7 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/Management.DataMigration/generate.cmd]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/Generated/DataMigrationServiceClient.cs (hunk 4 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/Management.DataMigration/Generated/DataMigrationServiceClient.cs]: I am generating using the latest Autorest (2.0.4284) from the public swagger where the Client does contain ApiVersion.  

autorest "https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datamigration/resource-manager/readme.md" --csharp --csharp-sdks-folder=F:\Git\Sync\azure-sdk-for-net\src\SDKs --version="latest" --reflect-api-versions

Is there a specific autorest version which I should be using?

Comment (dsgouda) [reply src/SDKs/DataMigration/Management.DataMigration/Generated/DataMigrationServiceClient.cs]: Confirmed this change is expected

File: src/SDKs/_metadata/.txt [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿Executing AutoRest command
+cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification//readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=F:\Git\Sync\azure-sdk-for-net\tools\..\src\SDKs\
+2018-09-04 21:06:14 UTC
+1) azure-rest-api-specs repository information
+GitHub fork: Azure
+Branch:      master
```
Comment (dsgouda) [inline src/SDKs/_metadata/.txt]: Please delete this file and regenerate the code

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj [hunk 10]
```diff
@@ -2,13 +2,13 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
   <PropertyGroup>
     <PackageId>DataMigration.Tests</PackageId>
-    <VersionPrefix>1.0.0-preview</VersionPrefix>
-    <AssemblyName>DataMigration.Tests</AssemblyName>
     <Description>DataMigration.Tests Class Library</Description>
+    <AssemblyName>DataMigration.Tests</AssemblyName>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: Please replace this with `<Version>1.0.0</Version>`

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 11]
```diff
@@ -17,6 +17,8 @@
 	</PropertyGroup>
 	<PropertyGroup>
 		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+		<FileVersion>0.4.0.0</FileVersion>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Remove these lines
`FileVersion` must be set in the `AssemblyInfo.cs` file

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a couple of comments

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj (hunk 10 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: Done

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj (hunk 11 above)
Comment (huang91shu) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Done

File: src/SDKs/_metadata/.txt (hunk 9 above)
Comment (huang91shu) [reply src/SDKs/_metadata/.txt]: Done.  Deleted ".txt" and reran generate.ps1 which updated "datamigration_resource-manager.txt"

**Review by dsgouda — APPROVED** [review]
LGTM
